src/journal/journald-server.c

```diff
@@ -1378,7 +1378,7 @@ int server_process_datagram(sd_event_source *es, int fd, uint32_t revents, void
 
         if (fd == s->syslog_fd) {
                 if (n > 0 && n_fds == 0)
-                        server_process_syslog_message(s, strstrip(s->buffer), ucred, tv, label, label_len);
+                        server_process_syslog_message(s, s->buffer + strspn(s->buffer, WHITESPACE), ucred, tv, label, label_len);
```

Comment:
To be honest I don't care much about this use-case after I found out that glibc's syslog() doesn't forward NULL bytes. I prepared another version of a patch that more-less implements your initial suggestion.